models/beethoven_x/fantom/beethoven_x_fantom_bpt_prices.sql

```diff
@@ -8,378 +8,12 @@
-            COALESCE(c2.token_in_price, u1.avg_price) AS token_in_price,
-            c2.token_out,
-            c2.out_in_norm_rate,
-            COALESCE(c2.token_out_price, u2.avg_price) AS token_out_price
-        FROM all_trades_calc_2 c2
-        LEFT JOIN unique_tx_token_price u1 ON u1.tx_hash = c2.tx_hash AND u1.token = c2.token_in
-        LEFT JOIN unique_tx_token_price u2 ON u2.tx_hash = c2.tx_hash AND u2.token = c2.token_out
-    ),
-
-    backfill_pricing_2 AS (
-        SELECT
-            block_time,
-            tx_hash,
-            bpt_address AS contract_address,
-            token_in,
-            COALESCE(token_in_price, (out_in_norm_rate * token_out_price)) AS token_in_price,
-            token_out,
-            COALESCE(token_out_price, (in_out_norm_rate * token_in_price)) AS token_out_price,
-            in_out_norm_rate,
-            out_in_norm_rate
-        FROM backfill_pricing_1
-    ),
-
-    trade_price_formulation AS (
-        SELECT
-            date_trunc('day', block_time) AS day,
-            contract_address,
-            approx_percentile(price, 0.5) FILTER (WHERE is_finite(price)) AS median_price
-        FROM (
-            SELECT block_time, contract_address, token_in_price AS price
-            FROM backfill_pricing_2 b2 WHERE b2.contract_address = b2.token_in
-            UNION
-            SELECT block_time, contract_address, token_out_price AS price
-            FROM backfill_pricing_2 b2 WHERE b2.contract_address = b2.token_out
-        )
-        GROUP BY 1, 2
-    ),
-
-    trade_price_formulation_2 AS(
-        SELECT
-            day,
-            contract_address,
-            CASE
-                WHEN median_price IS NOT NULL THEN median_price
-                WHEN LEAD(median_price) OVER(PARTITION BY contract_address ORDER BY day DESC) IS NOT NULL
-                THEN LEAD(median_price) OVER(PARTITION BY contract_address ORDER BY day DESC)
-                WHEN LAG(median_price) OVER(PARTITION BY contract_address ORDER BY day DESC) IS NOT NULL
-                THEN LAG(median_price) OVER(PARTITION BY contract_address ORDER BY day DESC)
-                ELSE approx_percentile(median_price, 0.5) OVER(
-                        PARTITION BY contract_address ORDER BY day
-                        ROWS BETWEEN 10 PRECEDING AND 10 FOLLOWING
-                    )
-            END AS median_price
-        FROM trade_price_formulation
-    ),
-
-    price_formulation AS(
-        SELECT
-            day,
-            contract_address,
-            median_price
-        FROM (
-            SELECT
-                day,
-                contract_address,
-                median_price,
-                AVG(median_price) OVER (PARTITION BY contract_address) AS avg_median_price
-            FROM trade_price_formulation
-        ) subquery
-        WHERE median_price < avg_median_price * 2 --removes outliers
-        GROUP BY 1, 2, 3
+{{ 
+    balancer_v2_compatible_bpt_prices_macro(
```

Comment:
Please check if the unmatched records are acceptable:
https://dune.com/queries/3869052

Reply:
unmatched records are more accurate with this new macro, according to beets' subgraph